pkg/logging/logging.go

```diff
@@ -170,16 +171,20 @@ func loggingProcessAdapter(ctx context.Context, driver Driver, dataStore string,
 	processLogFunc := func(reader io.Reader, dataChan chan string) {
 		defer wg.Done()
 		defer close(dataChan)
-		scanner := bufio.NewScanner(reader)
-		for scanner.Scan() {
-			if scanner.Err() != nil {
-				log.L.Errorf("failed to read log: %v", scanner.Err())
-				return
+		scanner := bufio.NewReader(reader)
+
+		var err error
+		var s string
+
+		for err == nil {
+			s, err = scanner.ReadString('\n')
+			if err != nil && err != io.EOF {
+				log.L.WithError(err).Error("failed to read log")
 			}
-			dataChan <- scanner.Text()
+
+			dataChan <- strings.TrimSuffix(s, "\n")
```

Comment:
TestLogsOutStreamsSeparated was failing if we would trim the suffix only when there is no error